edx_management_commands/management_commands/management/commands/manage_group.py

```diff
@@ -0,0 +1,140 @@
+"""
+Management command `manage_group` is used to idempotently create Django groups
+and set their permissions by name.
+"""
+
+from django.apps import apps
+from django.contrib.auth.models import Group, Permission
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.utils.translation import gettext as _
+
+
+class Command(BaseCommand):
+    # pylint: disable=missing-docstring
+
+    help = 'Creates the specified group, if it does not exist, and sets its permissions.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('group_name')
+        parser.add_argument('--remove', dest='is_remove', action='store_true')
+        parser.add_argument('-p', '--permissions', nargs='*', default=[])
+
+    def _handle_remove(self, group_name):
+
+        try:
+            Group.objects.get(name=group_name).delete()  # pylint: disable=no-member
+            self.stderr.write(_('Removed group: "{}"').format(group_name))
+        except Group.DoesNotExist:
+            self.stderr.write(_('Did not find a group with name "{}" - skipping.').format(group_name))
+
+    @transaction.atomic
+    def handle(self, group_name, is_remove, permissions=None, *args, **options):
+
+        if is_remove:
+            self._handle_remove(group_name)
+            return
+
+        old_permissions = []
+        group, created = Group.objects.get_or_create(name=group_name)  # pylint: disable=no-member
+
+        if created:
+            try:
+                # Needed for sqlite backend (i.e. in tests) because
+                # name.max_length won't be enforced by the db.
+                # See also http://www.sqlite.org/faq.html#q9
+                group.full_clean()
+            except ValidationError, exc:
+                # give a more helpful error
+                raise CommandError(
+                    _(
+                        'Invalid group name: "{group_name}". {messages}'
+                    ).format(
+                        group_name=group_name,
+                        messages=exc.messages[0]
+                    )
+                )
+            self.stderr.write(_('Created new group: "{}"').format(group_name))
+        else:
+            self.stderr.write(_('Found existing group: "{}"').format(group_name))
+            old_permissions = list(group.permissions.all())
+
+        new_permissions = self._resolve_permissions(permissions or [])
+
+        add_permissions = [p for p in new_permissions if p not in old_permissions]
+        remove_permissions = [p for p in old_permissions if p not in new_permissions]
```

Comment:
Why do we explicitly specify each permission to add/remove, rather than just setting `group.permissions.set(permissions)`, which does it all for us?